File: server/models/nation_ab.js
const mongoose = require("mongoose");
const NationSchemaAb = new mongoose.Schema({
  date: Date,
  ab_battles_mean: Number,
  ab_air_frags_per_battle: Number,
  nation: String,
  ab_battles_sum: Number,
  ab_win_rate: Number,
  ab_air_frags_per_death: Number,
  ab_ground_frags_per_battle: Number,
  ab_ground_frags_per_death: Number,
  ab_lower_br: Number,
  cls: String,
});

const ab_ranks_0 = mongoose.model("ab_ranks_0", NationSchemaAb, "ab_ranks_0");
const ab_ranks_1 = mongoose.model("ab_ranks_1", NationSchemaAb, "ab_ranks_1");
const ab_ranks_all = mongoose.model("ab_ranks_all", NationSchemaAb, "ab_ranks_all");
module.exports = { ab_ranks_0, ab_ranks_1, ab_ranks_all };

File: server/src/main/java/com/example/server/nationab/NationAb.java
package com.example.server.nationab;

import org.springframework.data.mongodb.core.mapping.Field;

import com.example.server.Nation;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public abstract class NationAb extends Nation {

    @Field("ab_lower_br")
    private Float lowerBr;
    @Field("ab_win_rate")
    private Float winRate;
    @Field("ab_battles_sum")
    private Float battleSum;
    @Field("ab_battles_mean")
    private Float battleMean;

    @Field("ab_ground_frags_per_battle")
    private Float groundFragPerBattle;
    @Field("ab_ground_frags_per_death")
    private Float groundFragPerDeath;

    @Field("ab_air_frags_per_battle")
    private Float airFragPerBattle;
    @Field("ab_air_frags_per_death")
    private Float airFragPerDeath;
}


